File: lib/app/Models/OrderDeliveryEacType.php
<?php
/**
 * OrderDeliveryEacType
 */
namespace app\Models;

/**
 * OrderDeliveryEacType
 * @description Тип кода подтверждения ЭАПП:  * `MERCHANT_TO_COURIER` — продавец передает код курьеру.  * `COURIER_TO_MERCHANT` — курьер передает код продавцу.  * `CHECKING_BY_MERCHANT` — продавец проверяет код на своей стороне.
 */
class OrderDeliveryEacType
{
    /**
     * Possible values of this enum
     */
    const MERCHANT_TO_COURIER = 'MERCHANT_TO_COURIER';

    const COURIER_TO_MERCHANT = 'COURIER_TO_MERCHANT';

    const CHECKING_BY_MERCHANT = 'CHECKING_BY_MERCHANT';

    /**
     * Gets allowable values of the enum
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::MERCHANT_TO_COURIER,
            self::COURIER_TO_MERCHANT,
            self::CHECKING_BY_MERCHANT
        ];
    }
}

File: src/main/kotlin/ympa_kotlin_wiremock_server/models/OrderDeliveryEacType.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

enum class OrderDeliveryEacType {
    @JsonProperty(value = "MERCHANT_TO_COURIER") MERCHANT_TO_COURIER,
    @JsonProperty(value = "COURIER_TO_MERCHANT") COURIER_TO_MERCHANT,
    @JsonProperty(value = "CHECKING_BY_MERCHANT") CHECKING_BY_MERCHANT,
}
